src/model/Vatom.js

```diff
@@ -96,4 +100,19 @@ export default class Vatom {
     // No match found, deny
     return false
   }
+
+  /** Checks if this vatom has an icon face */
+  containsIconFace () {
+    return !!this.faces.find(f => f.properties.constraints.platform === 'web' && f.properties.constraints.view_mode === 'icon')
```

Comment:
```suggestion
    return !!this.faces.find(f => (f.properties.constraints.platform === 'web' || f.properties.constraints.platform === 'generic') && f.properties.constraints.view_mode === 'icon')
```

Need to check for the generic platform constraint as well